libpromises/evalfunction.c

```diff
@@ -8607,7 +8573,7 @@ const FnCallType CF_FNCALL_TYPES[] =
                   FNCALL_OPTION_NONE, FNCALL_CATEGORY_COMM, SYNTAX_STATUS_NORMAL),
     FnCallTypeNew("datastate", CF_DATA_TYPE_CONTAINER, DATASTATE_ARGS, &FnCallDatastate, "Construct a container of the variable and class state",
                   FNCALL_OPTION_NONE, FNCALL_CATEGORY_UTILS, SYNTAX_STATUS_NORMAL),
-    FnCallTypeNew("difference", CF_DATA_TYPE_STRING_LIST, SETOP_ARGS, &FnCallSetop, "Returns all the unique elements of list arg1 that are not in list arg2",
+    FnCallTypeNew("difference", CF_DATA_TYPE_STRING_LIST, SETOP_ARGS, &FnCallSetop, "Returns all the unique elements of collected list or array or data container arg1 that are not in collected list or array or data container arg2",
```

Comment:
Yes, but that's what I mean. This is becoming the common case now, and definitely should be mentioned in the changelog, for instance. But maybe it doesn't make sense to mention it for every function, since the default is that it *does* support collection. And yes, this would involve changing the docs as well, perhaps inverting the concept, so that only functions that don't support collection are mentioned (but are there any, after this?).